### PR TITLE
Fix balance may confuse leader observer

### PR DESCRIPTION
**Pull request by yah01**
Signed-off-by: yah01 <yang.cen@zilliz.com>
#19430 
/kind bug

### CODE REVIEW

File: internal/querycoordv2/observers/leader_observer.go [hunk 1]
```diff
@@ -83,9 +83,11 @@ func (o *LeaderObserver) findNeedLoadedSegments(leaderView *meta.LeaderView, dis
...
-		consistentOnLeader := ok && node == s.Node
-		if consistentOnLeader || !o.target.ContainSegment(s.GetID()) {
+		version, ok := leaderView.Segments[s.GetID()]
+		consistentOnLeader := ok && version.NodeID == s.Node
+		if consistentOnLeader ||
+			ok && version.Version > s.Version ||
+			!o.target.ContainSegment(s.GetID()) {
 			continue
 		}
 		ret = append(ret, &querypb.SyncAction{
```
Comment (sunby) [inline internal/querycoordv2/observers/leader_observer.go:93]: SyncAction should have version

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**